### PR TITLE
Add e2e XLA JAX framework benchmarks and tools

### PR DESCRIPTION
**Pull request by pzread**
- Define `c2-standard-16` and `a2-highgpu-1g` benchmarks for T5 large JAX model
- Move and refactor the XLA JAX benchmark tool from https://github.com/iree-org/iree-samples/blob/main/iree-jax/benchmark/benchmark_model.py

### CODE REVIEW

**Comment by pzread** [comment]
Kindly ping : )

File: common_benchmark_suite/openxla/benchmark/comparative_suite/jax/benchmark_definitions.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import itertools
+
+from openxla.benchmark import def_types, unique_ids
+from openxla.benchmark.devices import gcp_devices
+from openxla.benchmark.comparative_suite import utils
+from openxla.benchmark.comparative_suite.jax import model_definitions, test_data_definitions
+
+T5_LARGE_JAX_A2_HIGHGPU_1G = utils.build_batch_benchmark_cases(
```
Comment (mariecwhite) [inline common_benchmark_suite/openxla/benchmark/comparative_suite/jax/benchmark_definitions.py:14]: Should this have FP32 and input size in the name? `T5_LARGE_FP32_JAX_523XI32_A2_HIGHGPU_1G`. It's long but it's possible that we'll have another benchmark with the same model but different input sequence length.

File: common_benchmark_suite/openxla/benchmark/comparative_suite/jax/test_data_definitions.py [hunk 2]
```diff
@@ -58,6 +58,6 @@
                     "https://storage.googleapis.com/iree-model-artifacts/jax/jax_models_0.4.10_1684396752/T5_LARGE/batch_${batch_size}/output_npy.tgz"
                 ))
     })
-T5_LARGE_FP32_JAX_512X1024XF32_BATCHES = utils.build_batch_model_test_data(
+T5_LARGE_FP32_JAX_512X1024XF32_OUTPUT_BATCHES = utils.build_batch_model_test_data(
```
Comment (mariecwhite) [inline common_benchmark_suite/openxla/benchmark/comparative_suite/jax/test_data_definitions.py:61]: For models, I think we've been following the convention: <model>_<data_type>_<framework>_<input_size>. For input and output data, I'm wondering if it should be prefixed with DATA_<input/output>_<model>_<data_type>_<framework>_<size>?

File: comparative_benchmark/xla_jax/benchmark_all.sh [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+declare -a CPU_BENCHMARK_NAMES=(
+  "models/T5_LARGE_FP32_JAX_512XI32_BATCH1/inputs/.+/expected_outputs/.+/target_devices/c2-standard-16"
+  "models/T5_LARGE_FP32_JAX_512XI32_BATCH16/inputs/.+/expected_outputs/.+/target_devices/c2-standard-16"
+  "models/T5_LARGE_FP32_JAX_512XI32_BATCH32/inputs/.+/expected_outputs/.+/target_devices/c2-standard-16"
+)
+
+if [ "${TARGET_DEVICE}" = "a2-highgpu-1g" ]; then
+    BENCHMARK_NAMES=("${GPU_BENCHMARK_NAMES[@]}")
+    ITERATIONS=50
```
Comment (mariecwhite) [inline comparative_benchmark/xla_jax/benchmark_all.sh:37]:
```suggestion
  BENCHMARK_NAMES=("${GPU_BENCHMARK_NAMES[@]}")
  ITERATIONS=50
```

File: comparative_benchmark/xla_jax/benchmark_all.sh [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+elif [ "${TARGET_DEVICE}" = "c2-standard-16" ]; then
+    BENCHMARK_NAMES=("${CPU_BENCHMARK_NAMES[@]}")
+    ITERATIONS=20
```
Comment (mariecwhite) [inline comparative_benchmark/xla_jax/benchmark_all.sh:40]:
```suggestion
  BENCHMARK_NAMES=("${CPU_BENCHMARK_NAMES[@]}")
  ITERATIONS=20
```

File: comparative_benchmark/xla_jax/run_benchmarks.py [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+    verbose: bool,
+):
+  name_pattern = re.compile(f"^{benchmark_name}$")
+  benchmarks = [
+      benchmark for benchmark in benchmark_definitions.BENCHMARKS
+      if name_pattern.match(benchmark.name)
+  ]
+
+  if not benchmarks:
+    raise ValueError(f'No benchmark matches "{benchmark_name}".')
```
Comment (mariecwhite) [inline comparative_benchmark/xla_jax/run_benchmarks.py:245]: Might be useful to print out the list of available benchmarks.

Comment (pzread) [reply comparative_benchmark/xla_jax/run_benchmarks.py]: Done.

**Review by mariecwhite — APPROVED** [review]

**Pull request merged**